src/figuresClasses.ts

```diff
@@ -1,19 +1,79 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('a, b and c must be > 0');
+    }
+
+    const sides = [this.a, this.b, this.c];
+    const longestSide = Math.max(...sides);
+
+    if (sides
+      .reduce((sum, side) => sum + side, 0) - longestSide <= longestSide) {
+      throw new Error('a + b should be greater than c');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
 
+    return Math.floor(Math.sqrt(p * (p - this.a)
+      * (p - this.b) * (p - this.c)) * 100) / 100;
+  }
```

Comment:
I would come up with something like that - to make it clear what happens on each step
```suggestion
    return Math.floor(
      Math.sqrt(
        p * (p - this.a) * (p - this.b) * (p - this.c)
      ) * 100) / 100
    };
```

Reply:
![image](https://github.com/mate-academy/ts_figures_classes/assets/127235037/4fc5c78d-0cef-4dfd-8a47-16ccd02b736b)

Reply:
@PANCH1811 so what happened when you followed these instructions?

Reply:
@silchyn a comma will destroy a mathematical expression, and adding braces- the linter asks for more braces, why?

Reply:
nvm, it somehow works with that comma...